tfjs-core/src/ops/spectral_ops.ts

```diff
@@ -56,6 +56,50 @@ function fft_(input: Tensor): Tensor {
   return ret.reshape(input.shape);
 }
 
+/**
+ * 2D fast Fourier transform.
+ *
+ * Computes the 2-dimensional discrete Fourier transform over the inner-most 2
+ * dimensions of input.
+ *
+ * ```js
+ * const real = tf.tensor2d([1, 2], [1, 2]);
+ * const imag = tf.tensor2d([1, 2], [1, 2]);
+ * const x = tf.complex(real, imag);
+ *
+ * x.fft2d().print();  // tf.spectral.fft2d(x).print();
+ * ```
+ * @param input The complex input to compute an fft2d over.
+ */
+/**
+ * @doc {heading: 'Operations', subheading: 'Spectral', namespace: 'spectral'}
+ */
+function fft2d_<T extends Tensor2D|Tensor3D>(input: T): T {
+  let input3D = input as Tensor3D;
+  let reshapedTo3D = false;
+
+  if (input.rank === 2) {
+    reshapedTo3D = true;
+    input3D = input.as3D(1, input.shape[0], input.shape[1]);
```

Comment:
If the input is 2d rank, we can just pass it to `fft`?

Reply:
I believe passing it to `fft` would just be the fft across the rows. Where _(I think)_ `fft2d` is the equivalent to fft across the rows, then columns. But definitely `fft(tensor2d) != fft2d(tensor2d)`

Reply:
I looked into it and `fft2d` is equivalent to `transpose(fft(transpose(fft(x))))`